### ISSUE TITLE
assertSame/assertNotSame broken after GITHUB-2296

### ISSUE THREAD

**Issue by vdiravka** [cbeust/testng#2486]
### TestNG Version
7.4.0 (7.3.0 worked fine)

### Expected behavior
`assertSame(object.clone(), object)` returns false

### Actual behavior
`assertSame(object.clone(), object)` returns true

### Description
assertSame/assertNotSame should use `==` operator, but not `equals` method to compare the same objects to identify they are the same object or different objects

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [x] IntelliJ
- [ ] NetBeans

### Test case sample
```
public void testAssertNotSame() {
    Contrived object = new Contrived(2);
    Contrived object2 = new Contrived(2);;
    Assert.assertNotSame(object2, object);
}
```

**Comment by javydreamercsw** [comment]
Was about to report this, thanks!

**Comment by Axinet** [comment]
Are there any plans to release 7.4.1 with this fix? Broken assertion seems like important enough part of test framework to release fix/patch?

**Comment by juherr** [comment]
@Axinet Sorry, no plan for a `7.4.1`. Do you have an issue with `7.3.0` which is resolved in `7.4.0`?

**Comment by Axinet** [comment]
It looked like I have, but after verification (I switched broken assertion to use AssertJ instead of TestNG assertion) it looks like for now 7.3.0 has no drawbacks in compare to 7.4.0. The issue we have is located somewhere elsewhere, although looked similar to one of the solved issues for 7.4.0.
**So for me it works with the old version good enough**, although I am still surprised that there is no patch planned. Based on thumb ups on my question some other people wonder about it too.

**Comment by juherr** [comment]
> I am still surprised that there is no patch planned

It is too much effort to deal with many branches. The fix will be included in the next release.
